app/javascript/components/pages/ApartmentEdit.js

```diff
@@ -1,9 +1,141 @@
 import React, { Component } from 'react'
 
+// form is added to apartment edit page and inputs are setting state on component
+
+
 export default class ApartmentEdit extends Component {
-  render() {
-    return (
-      <h3>ApartmentEdit</h3>
-    )
+    constructor(props) {
+        super(props)
+        this.state = {
+            form: {
+              street: '',
+              city: '',
+              state: '',
+              manager: '',
+              email: '',
+              price: '',
+              bedrooms: 0,
+              bathrooms: 0,
+              pets: '',
+            },
+          submitted: false,
+        }
+    }
+
+    handleChange = (e) => {
+        const { form } = this.state
+        form[e.target.name] = e.target.value
+        this.setState({ form })
+    }
+    handleSubmit = (e) => { 
+      this.props.updateApartment(this.state.form, this.props.apartment.id)
+      this.setState({ success: true })
```

Comment:
I don't see a key for success in state.  I think you initially had this as submitted for the key which is fine, but what does setting it to true do?  When submitted becomes true, the user should be redirected.

Reply:
This looks better, but you still need when success is true to trigger a redirect to the show page.  Take a look at this section in the syllabus: https://github.com/learn-academy-2022-delta/Syllabus/blob/main/cat-tinder/frontend/cat-update.md